src/codemagic/tools/app_store_connect.py

```diff
@@ -510,6 +513,50 @@ def register_device(
             platform=platform,
         )
 
+    @cli.action(
+        'register-devices-from-file',
+        BundleIdArgument.PLATFORM,
+        DeviceArgument.COMMON_DEVICE_NAME,
+        DeviceArgument.DEVICE_UDIDS_PATH,
+    )
+    def register_devices_from_file(
+        self,
+        platform: BundleIdPlatform,
+        device_name: str,
+        device_udids_path: pathlib.Path,
+        should_print: bool = True,
+    ) -> List[Device]:
```

Comment:
This action declaration / definition contradicts pretty much all existing principles we've had. There is no other action which reads input only from file and here too it would be expected that the user can register multiple devices using a single command by just passing multiple UDIDs instead of one. Reading from file (and environment variable) can be achieved with custom environment argument, which also does the validation.

Reply:
I thought to do just the way you proposed (add support for file to existing action and allow to pass multiple values). The problem was that the existing action raises an error for duplicate devices, while this action should ignore duplicates and move on

This would cause a major with a call like

```sh
app-store-connect register-device -u 00000000-000000000000000E -n iPhone --udids-path UDIDs.txt
```

This is the reason I created a separate action that would only accept UDIDs from a file, hence the `from-file` suffix.

As I had more time to think about it, I believe we can resolve this inconsistency by introducing a strict mode and have it enabled by default (otherwise the existing `register-device` would stop returning 1 exit code on error, it would be breaking change)

Reply:
As long as the default behaviour remains the same as it used to then we're good. Adding a boolean switch would work fine for that purpose.